debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerPartition.java

```diff
@@ -36,12 +45,12 @@ public boolean equals(Object obj) {
             return false;
         }
         final SqlServerPartition other = (SqlServerPartition) obj;
-        return Objects.equals(serverName, other.serverName);
+        return Objects.equals(serverName, other.serverName) && Objects.equals(databaseName, other.databaseName);
     }
 
     @Override
     public int hashCode() {
-        return serverName.hashCode();
+        return Objects.hash(serverName, databaseName);
```

Comment:
Let's call it once in the constructor and have a field storing this (constant) hash code.

Reply:
Done. I also added the database name to the return value of `getSourcePartition()`, if the multi-partition mode is enabled. The original patch didn't add it because there was no flag yet.